### ISSUE TITLE
std::complex< T > as attributes

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS2#1908]
Hi,

is there a specific reason why we do not allow `std::complex< T >` types as attributes in ADIOS2?
Those are primitive types and according to my tests in openPMD-api, using complex types as attributes was possible in ADIOS1.

Can we potentially just update the macros that specialize the overloads to allow complex types for attributes as well?

**Comment by ax3l** [comment]
The [specialization you need for `nlohmann::json`](https://github.com/nlohmann/json/issues/1510#issuecomment-471949812) reads:
```C++
#include <complex>

namespace std
{
    // std::complex handling
    template< class T > void to_json(nlohmann::json &j, const std::complex< T > &p) {
        j = nlohmann::json {p.real(), p.imag()};
    }

    template< class T > void from_json(const nlohmann::json &j, std::complex< T > &p) {
        p.real(j.at(0));
        p.imag(j.at(1));
    }
}
```

**Comment by ax3l** [comment]
Implemented via #1909 and #1922
